packages/flame/lib/src/geometry/shapes/circle.dart

```diff
@@ -0,0 +1,64 @@
+import 'dart:ui';
+
+import 'package:meta/meta.dart';
+
+import '../../extensions/vector2.dart';
+import '../../game/transform2d.dart';
+import 'shape.dart';
+
+/// The circle with a given [center] and [radius].
+@immutable
```

Comment:
If the `CircleComponent` should be backed by this later it needs to be mutable right?

Reply:
That's one of the things that needs to be decided.
If we want the objects to have `operator==` and `hashCode`, then the object has to be declared as `@immutable` (we have an analyzer rule for that). This wouldn't necessarily preclude the shape from being usable in many places -- just that we would need to create new objects on each modification.

Alternatively, we can make all shapes mutable, and then maybe instead of `operator==` declare a simple function `.equals(Shape other)`.

Immutable shapes are a bit easier to work with -- no need to check the validity of the shape on each mutation, certain properties that are expensive to calculate can be easily computed once and memorized, etc.

Reply:
I would love to have them immutable, but I don't think that would work performance wise in practice, since the hitboxes and shape components need to be able to change over time and we don't want to create shape objects in every tick when they are transforming.